### PR TITLE
Launchpad: Add domain ssl processing notification

### PR DESCRIPTION
**Pull request by jeyip**
### Proposed Changes

* Show a notification message underneath the domains preview noting that it is in the process of being set up

### Screenshots

Screen Shot 2022-10-21 at 12 20 06 AM

### Testing Instructions

* Check out this PR
* Create new site through tailored flow https://wordpress.com/hp-2022-tailored-flows/
* Make sure to switch out wordpress.com with calypso.localhost:3000
* Create a custom domain for the site
* Choose, at minimum, a personal plan ( so that we can make a custom domain the primary domain )
* Complete steps until at the Launchpad screen
* Verify that there is a domain ssl processing notification until ssl is active on the custom domain

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to https://github.com/Automattic/wp-calypso/issues/68967

### CODE REVIEW

**Comment by edanzer** [comment]
UPDATE: Issue above with free .wordpress.com domains showing the message has been resolved. 

- [x] Set up new site with custom domain and confirmed new message appears initially.
- [x] Set up free site (*.wordpress.com domain) to confirm the message does NOT show. From what I'm seeing, the new message still shows. I don't think that's expected. 
  - Just quickly checking the code to see what's happening, the sslStatus variable seems to be null for my new wordpress.com domain.

**Comment by jeyip** [comment]
> Set up free site (*.wordpress.com domain) to confirm the message does NOT show. From what I'm seeing, the new message still shows. I don't think that's expected.

Thanks for the patience with this @edanzer. I pushed a fix with tests to cover the null case

**Comment by agrullon95** [comment]
LGTM. 
- [x] Tested with existing free domain & paid domain : SSL notification message does not display in sidebar.
- [x] Tested with new paid domain on personal plan: SSL notification message displays in sidebar while custom domain SSL status is `pending`.

**Review by agrullon95 — APPROVED** [review]

**Review by sharpirate — APPROVED** [review]
Changes look good and this tests well.

### Testing
✔️ Notification is showing when using a pain domain with a personal plan while domain is pending
✔️ Notification is not showing when using a free domain

**Comment by Addison-Stavlo** [comment]
Regarding the notification - are we able to add the circle green check icon and corresponding text offset as shown in the similar design [here](https://github.com/Automattic/wp-calypso/issues/68967#issuecomment-1279018503) ?

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx [hunk 1]
```diff
@@ -71,7 +72,9 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep }: Sidebar
 		const { domain, isPrimary, isWPCOMDomain, sslStatus } = sidebarDomain;
 
 		[ siteName, topLevelDomain ] = getUrlInfo( domain );
-		showClipboardButton = isWPCOMDomain ? true : sslStatus === 'active' && isPrimary;
+
+		isDomainSSLProcessing = sslStatus !== 'active' && sslStatus !== null;
```
Comment (kirahsapong) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx]: [QUESTION] Out of curiosity (and aware these achieve the same aim) is there a style rule/reason for preferring this approach vs. `sslStatus && sslStatus !== 'active'` or `sslStatus && (sslStatus === 'pending' || sslStatus === 'disabled')`? 

Also, what are the general thoughts around using enums to maintain string values?

**Comment by kirahsapong** [comment]
Changes look good except for the green checkmark @Addison-Stavlo noted above. 

### Testing
[x] Notification shows with new custom domain
[x] Notification does not show with existing custom domain
[x] Notification does not show with any [site].wordpress.com domain
[ ]  Design matches mock-up

**Review by kirahsapong — APPROVED** [review]

**Comment by jeyip** [comment]
> Regarding the notification - are we able to add the circle green check icon and corresponding text offset as shown in the similar design https://github.com/Automattic/wp-calypso/issues/68967#issuecomment-1279018503 ?

I wasn't certain if we wanted to add the icon as well, but I'm happy to do so in a follow-up

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx (hunk 1 above)
Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx]: > Out of curiosity (and aware these achieve the same aim) is there a style rule/reason for preferring this approach vs. sslStatus && sslStatus !== 'active' or sslStatus && (sslStatus === 'pending' || sslStatus === 'disabled')?

Definitely no reason for preferring one approach over the other. I didn't give the conditional as much thought and spent more time figuring out the right name for the `isDomainSSLProcessing` variable ( to capture the intention of what I was doing ).

If anything, either of your two suggestions seems much clearer and I'm happy to swap the logic

Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx]: Fixed in https://github.com/Automattic/wp-calypso/pull/69317/commits/3cdf0a66d73777a4bf9d7e9b19540d2e5c8cd966

**Comment by edanzer** [comment]
Changes look good. Retested this after recent changes, and tests as expected. The issue I noted above with free .wordpress.com domains showing the new message is resolved. I also ran the unit tests, and they are all passing.

**Review by edanzer — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7596273

Thank you @jeyip for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7596273).

**Pull request merged**